src/components/ImageGallery/ImageGallery.tsx

```diff
@@ -0,0 +1,58 @@
+import { useKeenSlider } from "keen-slider/react";
+import ThumbnailPlugin from "./ThumbnailPlugin";
+
+interface Props {
+  images?: string[];
+  noProductImg?: string;
+}
+
+export const ImageGallery = ({ images, noProductImg }: Props) => {
+  const [sliderRef, instanceRef] = useKeenSlider<HTMLDivElement>({
+    initial: 0,
+    slides: {
+      perView: 1,
+      spacing: 10,
+    },
+  });
+
+  const [thumbnailRef] = useKeenSlider<HTMLDivElement>(
+    {
+      initial: 0,
+      slides: {
+        perView: 2,
+        spacing: 10,
+      },
+    },
+    [ThumbnailPlugin(instanceRef)],
+  );
+
+  return (
+    <div className="overflow-hidden">
+      {images ? (
+        <div className="flex flex-col gap-2">
+          <div ref={sliderRef} className="keen-slider h-full">
+            {images.map((i) => (
```

Comment:
It is better to use the full names, more readable `images => image`. In case you need the index, it will much more messy